app/models/source.rb

```diff
@@ -1,57 +1,54 @@
 class Source
   attr_accessor :path, :name
```

Comment:
This should either be changed to `attr_reader` or we define a new getter as:

```ruby
def path
  Pathname.new(@path)
end
```

as we are not guaranteed that `Source#path` will be a `Pathname` object if we allow it to be assigned to directly. Otherwise we'd have to audit the code everywhere to check all calls to`Source#path=`.

I mention this because there are a few places in the code below that makes the assumption that `Source#path` will always return a `Pathname` object.